listener/src/runtime_host.rs

```diff
@@ -35,7 +47,7 @@ impl RuntimeHostBuilder for RustRuntimeHostBuilder {
         _subgraph_id: SubgraphDeploymentId,
         _data_source: DataSource,
     ) -> Result<Self::Host, Error> {
-        Ok(RustRuntimeHost::new())
+        Ok(RustRuntimeHost::new(self.store.clone()))
```

Comment:
What is this `self.store.clone()` doing for us?

Reply:
It makes a copy of the store arc and passes it into the RustRuntimeHost. We need to pass a value rather than a reference since the RuntimeHost might run on a different thread and otherwise multiple of these threads might be accessing the same instance. 
If we just passed by value without clone, we could only ever build one RustRuntimeHost (which might be what we are doing atm but feels like an unnecessary restriction).